valhalla/baldr/landmark.h

```diff
@@ -0,0 +1,187 @@
+  bar = 13,
+  // this could be very very big like a college campus, i think we should remove this
+  hospital = 14,
+  // same as bar could be just a building need at least a name to have a chance of seeing it
+  pub = 16,
+  // generic, could be many many types of clinics also often there are multiple of these located in
+  // same "complex", you might have a dentist, orthodontist, ear nose throat and eye specialists all
+  // next to each other.
+  clinic = 17,
+  // on the fence about this one, to me this would be in the same category as "museum" or something
+  // where its so specialized maybe it doesnt need a name. what keeps me from say ing that is you
+  // often cant tell from the outside what function these amenities serve until you read the name
+  theatre = 16,
+  // this could work without a name in my opinion, they often have a board out front that shows what
+  // movies are playing which would probably remove the need for a name
+  cinema = 17,
+  // these are usually quite large as well but not usually multiple buildings, how often are they
+  // located near one another though? if its often perhaps we need a name to differentiate between
+  // this casino and the next one down on the next block? maybe that argument is a good argument for
+  // everything has to have a name?
+  casino = 18,
+};
+
+inline LandmarkType string_to_landmark_type(const std::string& s) {
+  static const std::unordered_map<std::string, LandmarkType> string_to_landmark_type =
+      {{"restaurant", LandmarkType::restaurant},
+       {"fast_food", LandmarkType::fast_food},
+       {"cafe", LandmarkType::cafe},
+       {"fuel", LandmarkType::fuel},
+       {"bank", LandmarkType::bank},
+       {"pharmacy", LandmarkType::pharmacy},
+       {"kindergarten", LandmarkType::kindergarten},
+       {"bar", LandmarkType::bar},
+       {"hospital", LandmarkType::hospital},
+       {"post_office", LandmarkType::post_office},
+       {"pub", LandmarkType::pub},
+       {"clinic", LandmarkType::clinic},
+       {"police", LandmarkType::police},
+       {"fire_station", LandmarkType::fire_station},
+       {"car_wash", LandmarkType::car_wash},
+       {"theatre", LandmarkType::theatre},
+       {"cinema", LandmarkType::cinema},
+       {"casino", LandmarkType::casino}};
+
+  auto it = string_to_landmark_type.find(s);
+  if (it == string_to_landmark_type.cend()) {
+    throw std::runtime_error("unknown landmark type");
+  }
+  return it->second;
+}
+
+inline uint64_t encode_lnglat(double lng, double lat) {
+  return (((uint64_t(lng * 1e7) + uint64_t(180 * 1e7)) & ((1ull << 32) - 1)) << 31) |
+          ((uint64_t(lat * 1e7) + uint64_t(90 * 1e7)) & ((1ull << 31) - 1));
+}
+
+inline std::pair<double, double> decode_lnglat(uint64_t location) {
+  double lng = (int64_t((location >> 31) & ((1ull << 32) - 1)) - 180 * 1e7) * 1e-7;
+  double lat = (int64_t(location & ((1ull << 31) - 1)) - 90 * 1e7) * 1e-7;
+
+  return std::make_pair(lng, lat);
+}
```

Comment:
@kevinkreiser @nilsnolde I took `pointll.h` as reference and updated the conversions of lng and lat and now they work as expected. But the returned string size is 8 bytes now. I didn't success on 4 bytes.